common/src/main/java/com/wynntils/services/itemfilter/LevelItemFilterFactory.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.services.itemfilter;
+
+import static java.lang.Integer.parseInt;
+
+import com.wynntils.utils.type.ErrorOr;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.client.resources.language.I18n;
+
+public class LevelItemFilterFactory implements ItemFilterFactory {
+    private static final Pattern LEVEL_RANGE_PATTERN = Pattern.compile("^(\\d+)(?:-(\\d+))?$");
+
+    @Override
+    public ErrorOr<LevelItemFilter> create(String inputString) {
+        int minLevel;
+        int maxLevel;
+        Matcher matcher = LEVEL_RANGE_PATTERN.matcher(inputString);
+        if (!matcher.find()) {
+            return ErrorOr.error(I18n.get(getI18nKey() + ".invalidRange", inputString));
+        }
+
+        try {
+            minLevel = parseInt(matcher.group(1));
+            maxLevel = matcher.group(2) == null ? minLevel : parseInt(matcher.group(2));
+        } catch (NumberFormatException ignore) {
+            return ErrorOr.error(I18n.get(getI18nKey() + ".invalidRange", inputString));
+        }
+
+        if (minLevel > maxLevel) {
+            return ErrorOr.error(I18n.get(getI18nKey() + ".maxLessThanMin", inputString));
+        }
+
+        return ErrorOr.of(new LevelItemFilter(minLevel, maxLevel));
+    }
+
+    @Override
+    public String getKeyword() {
+        return "lvl";
+    }
+
+    @Override
+    public String getI18nKey() {
+        return "feature.wynntils.itemFilter.level";
```

Comment:
You can use a little bit of reflection magic to have this automatically. Check out `com.wynntils.core.consumers.functions.Function#getTranslatedName` and try to copy what it does.

Reply:
@magicus I would love to have your opinion on that. The way I see it is it's always better not to have duplicated code, so I should have an abstract superclass implementing `ItemFilterFactory` handling the `getTranslationKey()` and having a constructor prototype like  `AsbtractItemFilterFactory(String name, List aliases)`.

That's the way I would have done it but now it seems kind of wrong to me with all the suggestion you made in this PR.

Reply:
I think you mentioned moving to a new name, maybe it is time to move away from being an interface as well.. I know we suggested it first..